### ISSUE TITLE
Add the capability to register post-to hooks

### ISSUE THREAD

**Issue by hugary1995** [applied-material-modeling/neml2#83]
After a Model is sent to a different device, all of its registered parameter will be sent to that device as well.
However, derived quantities are not, for example the elasticity tensor as a function of young's modulus and Poisson's ratio.
We need an interface to register post-to() hooks.

**Comment by hugary1995** [comment]
This is no longer needed after #124
